src/main/scala/sample/stream/GroupLogFile.scala

```diff
@@ -29,16 +31,13 @@ object GroupLogFile {
         case other                  => "OTHER"
       }.
       // write lines of each group to a separate file
-      runForeach {
+      mapAsync(parallelism = 4, {
```

Comment:
Of course. I had it changed to five after thinking about it, but somehow forgot to push it. I will do a quick PR to change it to five.